Running `setIsAsync(true)` on `ArrowFunction` incorrectly inserts `async` keyword

<!-- If you are contributing this bug fix then please read the instructions in CONTRIBUTING.md -->

**Describe the bug**

Version: 24.0.0 (latest)

<!-- A description of what the bug is. -->

Using `setIsAsync(true)` on an `ArrowFunction` (which indirectly calls `addModifier`) can result in an error due to incorrect insertion of the `async` keyword.

**To Reproduce**

<!-- Identify the problem and submit some reproduction code. Prune the reproduction code to remove needless details. State the actual behaviour. -->

Self-contained reproduction file:

```js
import { Node, Project } from "ts-morph";

const project = new Project();
const fileText = `const f = a => a * 2;`;
const sourceFile = project.createSourceFile("main.js", fileText);

sourceFile.forEachDescendant((node) => {
  if (Node.isArrowFunction(node)) {
    node.setIsAsync(true);
  }
});
```

This results in:

```
ManipulationError: Manipulation error: A syntax error was inserted.

main.js:1:13 - error TS1005: ',' expected.

1 const f = a async => a * 2;
              ~~~~~
main.js:1:19 - error TS1005: ';' expected.

1 const f = a async => a * 2;
                    ~~

Error replacing tree: The children of the old and new trees were expected to have the same count (2:3).

-- Details --
Path: repro/main.js
Text: "const f = a async => a * 2;"
Stack: Error: Error replacing tree: The children of the old and new trees were expected to have the same count (2:3).
    at #handleChildren (repro/node_modules/ts-morph/dist/ts-morph.js:1469:19)
    at ParentFinderReplacementNodeHandler.handleNode (repro/node_modules/ts-morph/dist/ts-morph.js:1463:33)
    at ParentFinderReplacementNodeHandler.handleNode (repro/node_modules/ts-morph/dist/ts-morph.js:1620:19)
    at doManipulation (repro/node_modules/ts-morph/dist/ts-morph.js:2364:21)
    at insertIntoParentTextRange (repro/node_modules/ts-morph/dist/ts-morph.js:2398:5)
    at ArrowFunction.addModifier (repro/node_modules/ts-morph/dist/ts-morph.js:9175:13)
    at ArrowFunction.toggleModifier (repro/node_modules/ts-morph/dist/ts-morph.js:9152:22)
    at ArrowFunction.setIsAsync (repro/node_modules/ts-morph/dist/ts-morph.js:2884:18)
    at file://repro/arrow.mjs:9:10
    at nodeCallback (repro/node_modules/ts-morph/dist/ts-morph.js:3345:33)
    at throwError (repro/node_modules/ts-morph/dist/ts-morph.js:2380:19)
    at doManipulation (repro/node_modules/ts-morph/dist/ts-morph.js:2374:13)
    at insertIntoParentTextRange (repro/node_modules/ts-morph/dist/ts-morph.js:2398:5)
    at ArrowFunction.addModifier (repro/node_modules/ts-morph/dist/ts-morph.js:9175:13)
    at ArrowFunction.toggleModifier (repro/node_modules/ts-morph/dist/ts-morph.js:9152:22)
    at ArrowFunction.setIsAsync (repro/node_modules/ts-morph/dist/ts-morph.js:2884:18)
    at file://repro/arrow.mjs:9:10
    at nodeCallback (repro/node_modules/ts-morph/dist/ts-morph.js:3345:33)
    at repro/node_modules/ts-morph/dist/ts-morph.js:3386:37
    at VariableDeclaration.forEachChild (repro/node_modules/ts-morph/dist/ts-morph.js:3325:37) {
  filePath: 'repro/main.js',
  oldText: 'const f = a => a * 2;',
  newText: 'const f = a async => a * 2;'
}

Node.js v20.18.0
```

**Expected behavior**

<!-- A description of what you expected to happen. -->

I would expect the resulting file to be `const f = async a => a * 2;`.

--- a/packages/ts-morph/src/compiler/ast/base/ModifierableNode.ts
+++ b/packages/ts-morph/src/compiler/ast/base/ModifierableNode.ts
@@ -161,6 +161,8 @@ export function ModifierableNode<T extends Constructor<ModifierableNodeExtension
         function getInitialInsertPos() {
           if (modifiers.length > 0)
             return modifiers[0].getStart();
+          if (node.getKind() === SyntaxKind.ArrowFunction)
+            return node.getStart();
           for (const child of node._getChildrenIterator()) {
             // skip over any initial syntax lists (ex. decorators) or js docs
             if (child.getKind() === SyntaxKind.SyntaxList || ts.isJSDocCommentContainingNode(child.compilerNode))

--- a/packages/ts-morph/src/tests/compiler/ast/base/asyncableNodeTests.ts
+++ b/packages/ts-morph/src/tests/compiler/ast/base/asyncableNodeTests.ts
@@ -1,6 +1,6 @@
-import { nameof } from "@ts-morph/common";
+import { nameof, SyntaxKind } from "@ts-morph/common";
 import { expect } from "chai";
-import { AsyncableNode, FunctionDeclaration } from "../../../../compiler";
+import { AsyncableNode, FunctionDeclaration, Node, VariableStatement } from "../../../../compiler";
 import { AsyncableNodeStructure } from "../../../../structures";
 import { getInfoFromText } from "../../testHelpers";
 
@@ -43,8 +43,10 @@ describe("AsyncableNode", () => {
 
   describe(nameof<AsyncableNode>("setIsAsync"), () => {
     function doTest(text: string, value: boolean, expected: string) {
-      const { firstChild, sourceFile } = getInfoFromText<FunctionDeclaration>(text);
-      firstChild.setIsAsync(value);
+      let { firstChild, sourceFile } = getInfoFromText(text);
+      if (firstChild.getKind() === SyntaxKind.VariableStatement)
+        firstChild = (firstChild as VariableStatement).getDeclarations()[0].getInitializerOrThrow();
+      (firstChild as any as AsyncableNode).setIsAsync(value);
       expect(sourceFile.getText()).to.equal(expected);
     }
 
@@ -54,6 +56,12 @@ describe("AsyncableNode", () => {
 
     it("should set as not async when async", () => {
       doTest("async function Identifier() {}", false, "function Identifier() {}");
+    });
+
+    it("should handle arrow function", () => {
+      doTest("const f = a => a * 2;", true, "const f = async a => a * 2;");
+      doTest("const f = async a => a * 2;", true, "const f = async a => a * 2;");
+      doTest("const f = async a => a * 2;", false, "const f = a => a * 2;");
     });
   });
 

